org.eclipse.scanning.connector.epicsv3/src/org/eclipse/scanning/connector/epicsv3/EpicsV3DynamicDatasetConnector.java

```diff
@@ -299,6 +274,56 @@ public boolean refreshShape() {
 		return false;
 	}
 	
+	private int getDataSize() {
+		String currentWidthPV;
+		String currentHeightPV;
+		String rgbChannelsPV;
+
+		if (colourMode.equals(monoColourMode)) {
+
+			width = dim0;		currentWidthPV = dim0PV;
+			height = dim1;		currentHeightPV = dim1PV;
+			rgbChannels = 1;	rgbChannelsPV = dim2PV;
+
+		} else {
+			if (colourMode.equals("RGB2")) {
+
+				width = dim0;		currentWidthPV = dim0PV;
+				height = dim2;		currentHeightPV = dim2PV;
```

Comment:
This rule is particularly annoying as complying with it actually makes the code less readable in cases like this.